frontend/src/main/java/com/google/tripmeout/frontend/service/PlaceService.java

```
@@ -0,0 +1,16 @@
+package com.google.tripmeout.frontend.service;
+
+import com.google.maps.errors.ApiException;
+import com.google.tripmeout.frontend.PlaceVisitModel;
+import java.io.IOException;
+import java.lang.InterruptedException;
+
+public interface PlaceService {
+  /**
+   * check that a placeId is valid based on the Places API
+   *
+   * @param placeId the id of the place to vaildate
+   */
+  boolean validatePlaceId(String placesApiPlaceId)
+      throws ApiException, InterruptedException, IOException;
```

This interface should be agnostic to the underlying implementation. In particular, we should not expose Maps API exceptions. Throwing interruption exceptions _might_ make sense if we expect operations to potentially block for a while.

Further, if validation failure should _always_ result in a client error, consider throwing an exception if the validation comes back negative rather than returning a boolean (which is easy to forget to check). I would keep the boolean if you think there will be situations where we want to change behavior following a validation failure but _not_ immediately stop computation and return an error to the client (seems unlikely, but possible).